Autodesk/synthesis
# Core Meta Panel & Gamepiece Configuration

- Moved all remaining general and advanced options to be in one tab.
- Everything is now structured like #995 for easy (er) refactoring of `ConfigCommand.py`.
- Integrated #1014.
- Made some small improvements to the way auto calculating robot weight works.
  - Now a checkbox instead of a button.
  - Choosing a different unit will now update he value input in real time.
- Also refactored gamepiece config options to be structured like #995.
- Switching between `Static` and `Dynamic` export mode will toggle the visibility of the joint config tab, the gampiece config tab, and the dynamic export specific options.

I am still not 100% sure how I feel about structuring the UI in this way. Seems better than what we were doing before but I still think it can be improved upon.

I combined both these JIRA tickets to avoid any downtime in the gamepiece configuration functionality as the meta panel task would have overridden / deprecated some of the field configuration options.

### To Test

This is a rather large PR here are somethings to make sure still work:
1. Exporting a robot normally.
2. Exporting a field normally.
3. Selection of game pieces working as expected.

Most of these changes were structural so everything should look and feel like it always has.

> [!important]
> Blocked by:
> - [x] #1014.
>   - Need to move friction data into the general tab for dynamic exports.
> - [x] #1045.
>   - Types file for unit typing and conversions.

[JIRA Issue - Meta Panel](https://jira.autodesk.com/browse/AARD-1683)
[JIRA Issue - Gamepiece Config](https://jira.autodesk.com/browse/AARD-1769)
